thread-utils.c

```diff
@@ -18,6 +18,67 @@
 #  endif
 #endif
 
+#ifdef GIT_WINDOWS_NATIVE
+/// defines the GetLogicalProcessorInformationEx function
+typedef BOOL (*WINAPI glpie_t)(
+	LOGICAL_PROCESSOR_RELATIONSHIP,
+	PSYSTEM_LOGICAL_PROCESSOR_INFORMATION_EX,
+	PDWORD);
```

Comment:
Please move this code to `compat/win32/` and use `lazyload.h` (imitate existing code that [declares](https://github.com/git-for-windows/git/blob/ad0bbfffa543db6979717be96df630d3e5741331/compat/mingw.c#L2807-L2808) and [initializes](https://github.com/git-for-windows/git/blob/ad0bbfffa543db6979717be96df630d3e5741331/compat/mingw.c#L2812) the function pointer).